CoreText/include/CoreText/CTFont.h

```diff
@@ -16,91 +16,280 @@ FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+    kCTFontTableKern = 'kern',
+    kCTFontTableKerx = 'kerx',
+    kCTFontTableLcar = 'lcar',
+    kCTFontTableLoca = 'loca',
+    kCTFontTableMaxp = 'maxp',
+    kCTFontTableMort = 'mort',
+    kCTFontTableMorx = 'morx',
+    kCTFontTableName = 'name',
+    kCTFontTableOpbd = 'opbd',
+    kCTFontTablePost = 'post',
+    kCTFontTablePrep = 'prep',
+    kCTFontTableProp = 'prop',
+    kCTFontTableSbit = 'sbit',
+    kCTFontTableSbix = 'sbix',
+    kCTFontTableTrak = 'trak',
+    kCTFontTableVhea = 'vhea',
+    kCTFontTableVmtx = 'vmtx'
+};
 typedef FourCharCode CTFontTableTag;
 
+typedef enum {
+    kCTFontTableOptionNoOptions = 0,
+    kCTFontTableOptionExcludeSynthetic = (1 << 0) // Deprecated
+} CTFontTableOptions;
+
+typedef enum {
+    kCTFontOptionsDefault = 0,
+    kCTFontOptionsPreventAutoActivation = 1 << 0,
+    kCTFontOptionsPreferSystemFont = 1 << 2,
+    kCTFontOptionsPreventAutoDownload = 1 << 1
+} CTFontOptions;
+
 CF_IMPLICIT_BRIDGING_ENABLED
 
-CORETEXT_EXPORT CTFontRef CTFontCreateWithGraphicsFont(
-        CGFontRef cgFont, CGFloat size, CGAffineTransform *xform,
-        /* CTFontDescriptorRef */ void *attributes);
+CORETEXT_EXPORT CTFontRef CTFontCreateWithName(CFStringRef name, CGFloat size,
+                                               const CGAffineTransform *matrix);
+CORETEXT_EXPORT CTFontRef CTFontCreateWithNameAndOptions(
+        CFStringRef name, CGFloat size, const CGAffineTransform *matrix,
+        CTFontOptions options);
+CORETEXT_EXPORT CTFontRef CTFontCreateWithFontDescriptor(
+        CTFontDescriptorRef descriptor, CGFloat size,
+        const CGAffineTransform *matrix);
+CORETEXT_EXPORT CTFontRef CTFontCreateWithFontDescriptorAndOptions(
+        CTFontDescriptorRef descriptor, CGFloat size,
+        const CGAffineTransform *matrix, CTFontOptions options);
 CORETEXT_EXPORT CTFontRef CTFontCreateUIFontForLanguage(
         CTFontUIFontType uiFontType, CGFloat size, CFStringRef language);
-CORETEXT_EXPORT CFStringRef CTFontCopyFullName(CTFontRef self);
+CORETEXT_EXPORT CTFontRef CTFontCreateCopyWithAttributes(
+        CTFontRef font, CGFloat size, const CGAffineTransform *matrix,
+        CTFontDescriptorRef attributes);
+CORETEXT_EXPORT CTFontRef CTFontCreateCopyWithSymbolicTraits(CTFontRef font, CGFloat size, const CGAffineTransform *matrix, CTFontSymbolicTraits symTraitValue, CTFontSymbolicTraits symTraitMask);
+CORETEXT_EXPORT CTFontRef CTFontCreateCopyWithFamily(CTFontRef font, CGFloat size, const CGAffineTransform *matrix, CFStringRef family);
+CORETEXT_EXPORT CTFontRef CTFontCreateForString(CTFontRef currentFont, CFStringRef string, CFRange range);
+CORETEXT_EXPORT CTFontRef CTFontCreateForStringWithLanguage(CTFontRef currentFont, CFStringRef string, CFRange range, CFStringRef language);
+
+CORETEXT_EXPORT CTFontDescriptorRef CTFontCopyFontDescriptor(CTFontRef font);
+CORETEXT_EXPORT CFTypeRef CTFontCopyAttribute(CTFontRef font,
+                                              CFStringRef attribute);
```

Comment:
This should be one line, not two
```suggestion
CORETEXT_EXPORT CFTypeRef CTFontCopyAttribute(CTFontRef font, CFStringRef attribute);
```